test.js

```diff
@@ -130,16 +134,82 @@ test('stripFinalNewline: false', async t => {
 	t.is(stdout, 'foo\n');
 });
 
-test('stripFinalNewline on failure', async t => {
-	const {stderr} = await t.throwsAsync(execa('noop-throw', ['foo'], {stripFinalNewline: true}));
-	t.is(stderr, 'foo');
+test('execa() with .kill after it with SIGKILL should kill cleanly', async t => {
+	const subprocess = execa('node', ['fixtures/no-killable'], {
+		stdio: ['ipc']
+	});
+
+	await waitForProcessLaunch(subprocess);
+
+	subprocess.kill('SIGKILL');
+
+	const {signal} = await t.throwsAsync(subprocess);
+	t.is(signal, 'SIGKILL');
 });
 
+// Windows doesn't support sending signals. No re-emulates them down to SIGKILL
+if (process.platform !== 'win32') {
+	test('execa() with .kill after it with SIGTERM should not kill (no retry)', async t => {
+		const subprocess = execa('node', ['fixtures/no-killable'], {
+			stdio: ['ipc']
+		});
+
+		await waitForProcessLaunch(subprocess);
+
+		subprocess.kill('SIGTERM', {
+			retry: false,
+			retryAfter: 50
+		});
+
+		t.true(isRunning(subprocess.pid));
+		subprocess.kill('SIGKILL');
+	});
+}
+
+// Windows doesn't support sending signals. No re-emulates them down to SIGKILL
+if (process.platform !== 'win32') {
+	test('execa() with .kill after it with SIGTERM should kill after 50 ms with SIGKILL', async t => {
+		const subprocess = execa('node', ['fixtures/no-killable'], {
+			stdio: ['ipc']
+		});
+
+		await waitForProcessLaunch(subprocess);
+
+		subprocess.kill('SIGTERM', {
+			retryAfter: 50
+		});
+
+		const {signal} = await t.throwsAsync(subprocess);
+		t.is(signal, 'SIGKILL');
+	});
+}
+
+// Windows doesn't support sending signals. No re-emulates them down to SIGKILL
+if (process.platform !== 'win32') {
+	test('execa() with .kill after it with nothing (undefined) should kill after 50 ms with SIGKILL', async t => {
+		const subprocess = execa('node', ['fixtures/no-killable'], {
+			stdio: ['ipc']
+		});
+
+		await waitForProcessLaunch(subprocess);
+
+		subprocess.kill();
+
+		const {signal} = await t.throwsAsync(subprocess);
+		t.is(signal, 'SIGKILL');
+	});
+}
+
 test('stripFinalNewline in sync mode', t => {
 	const {stdout} = execa.sync('noop', ['foo'], {stripFinalNewline: true});
 	t.is(stdout, 'foo');
 });
 
+test('stripFinalNewline on failure', async t => {
```

Comment:
After the rebase, the tests are now in the middle of the `stripFinalNewline` tests. Could you put it just before them?

Reply:
Done: d793df20c3e1ca071510849b295f16257306ee1e